lib/Dialect/Torch/Transforms/RefineTypes.cpp

```diff
@@ -1073,6 +1078,21 @@ ChangeResult TypeAnalyzer::visitConstantTensorAllocOp(OpTy op) {
   return getLatticeElement(op.getResult()).join(knowledge);
 }
 
+// Convert input tensor type to the given `dtype`.
+ChangeResult TypeAnalyzer::visitAtenToDtypeOp(
+    AtenToDtypeOp op, ArrayRef<LatticeElement<ValueKnowledge> *> operands) {
+  auto input = operands[0]->getValue();
+  auto knowledge =
+      ValueKnowledge::getNotNonePessimisticValueState(op->getContext());
+  if (!input.hasSizes)
```

Comment:
We would still want to fill in the dtype even when `!input.hasSizes`.  Could change the below to
```
if (input.sizes)
   knowledge.sizes = input.sizes;
```